syntax = "proto2";

package android.automotive.surround_view.object_detection;

// TODO(b/162017903): Update DetectedObjects to include multiple DetectedObject
// and each object has label, score, bounding_box, etc.
message DetectedObjects {
  // Detected object string label
  repeated string label = 1;

  // Detected object confidence score
  repeated float score = 2;

  // Detected object bounding box
  repeated BoundingBox bounding_box = 3;

  // The timestamp (in microseconds) *at which* this object was
  // created/detected.
  repeated int64 timestamp_usec = 4;

  // Camera id, designed for surround view to distinguish which camera's frame
  // is used to generate the detected objects.
  optional int32 camera_id = 5;
}

message BoundingBox {
  // Each bounding box have 4 corners. The relative position of corners should
  // be following.
  // (xmin, ymin)       (xmax, ymin)
  //   corner1-------------corner3
  //       |                  |
  //       |                  |
  //       |                  |
  //   corner2-------------corner4
  // (xmin, ymax)       (xmax, ymax)
  optional CornerPoint corner1 = 1;
  optional CornerPoint corner2 = 2;
  optional CornerPoint corner3 = 3;
  optional CornerPoint corner4 = 4;
}

message CornerPoint {
  // corner coordinates are absolute value within input image dimension
  optional float x = 1;
  optional float y = 2;
}
